### PR TITLE
Bump actions/cache from v1 to v2

### PR DESCRIPTION
**Pull request by imbsky**
We just released v2. That includes a lot of improvements.
https://github.com/actions/cache/releases/tag/v2.0.0

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -26,22 +26,22 @@ jobs:
...
+        uses: actions/cache@v2
         with:
           path: ~/.cargo/bin
           key: ${{ runner.os }}-cargo-bin-${{ hashFiles('**/Cargo.lock') }}
       - name: Cache Cargo Git
-        uses: actions/cache@v1
+        uses: actions/cache@v2
         with:
           path: ~/.cargo/git
           key: ${{ runner.os }}-cargo-git-${{ hashFiles('**/Cargo.lock') }}
```
Comment (Kixiron) [inline .github/workflows/ci.yml:42]: Since v2 allows multi-path caching, can you combine these into one single cache?

Comment (imbsky) [reply .github/workflows/ci.yml]: Yep, you're right!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -25,27 +25,20 @@ jobs:
...
+            ~/.cargo/registry
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
       - name: Cache Cargo Build
-        uses: actions/cache@v1
+        uses: actions/cache@v2
         with:
           path: target
           key: ${{ runner.os }}-cargo-build-target-${{ env.CURRENT_RUSTC_VERSION }}-${{ hashFiles('**/Cargo.lock') }}
-          restore-key: |
+          restore-keys: |
```
Comment (imbsky) [inline .github/workflows/ci.yml:41]: It's not directly related to this PR, but it was wrong.

**Comment by jyn514** [comment]
Thanks for the PR!

**Pull request merged**